### DISCUSSION TITLE
Kernel launch parameters for `thrust::for_each` (and other algorithms)

### DISCUSSION THREAD

**Discussion by mz-zoox** [NVIDIA/cccl#1052] [Thrust]
Is there a way to customize the kernel launch parameter for thrust algorithms? `thrust::for_each` always launches 512 CUDA threads per block.   I am wondering if it something user can customize for performance tuning?

Also related to launch parameters, but possible a new topic entirely. Is it possible to use shared memory in functor passed to `thrust::for_each` and if so, is dynamic shared memory possible and how to specify the size?

**Comment by alliepiper** [comment]
Thrust does not expose block size or shmem allocations to users, these are always hardcoded for each algorithm/architecture. You'll need to write a custom CUDA kernel outside of Thrust if you want this level of control.
